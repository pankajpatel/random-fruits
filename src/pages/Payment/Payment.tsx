import { useEffect, useState } from "react";
import { FormattedMessage } from "react-intl";
import styled from "styled-components";
import { Badge } from "../../components/Badge";
import { Spinner } from "../../components/Spinner";
import { usePayment } from "../../hooks/usePayment";

const Code = styled.code`
  font-weight: normal;
  background-color: #eee;
  border-radius: 0.25rem;
  padding: 0.25rem;
  font-style: italic;
`

export const Payment = ({ id }: { id: Payment["id"] }): JSX.Element => {
  const [payment, setPayment] = useState<Payment | undefined>();
  const {data, isLoading, isError} = usePayment(id)
  useEffect(() => {
    data && setPayment(data);
  }, [data])
  return (
    <section>
      <h2>
        <FormattedMessage id="pages.payment.title" defaultMessage="Payment" />{" "}
        <Code>{id}</Code>
      </h2>
      <hr />
      {isLoading && <Spinner />}
      {isError && (
        <p>
          Failed to load payment <code>{id}</code> details
        </p>
      )}
      {!isLoading && !isError && payment && (
        <div>
          <p>{payment.id}</p>
          <p>{payment.amount}</p>
          <p>{payment.created}</p>
          <p>{payment.customer_name}</p>
          <Badge type="info">{payment.status}</Badge>
        </div>
      )}
    </section>
  );
};

export default Payment
