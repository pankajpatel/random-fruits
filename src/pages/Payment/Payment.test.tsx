import { render } from "@testing-library/react";
import { Payment } from "./Payment";

import { mockEndpoint, MockableHTTPMethods } from "@app/testing/api-mocks";
import { withProviders } from "@app/testing/provider-setup";

import mockPaymentJSON from "@app/fixtures/payment.json";

const mockPayment = {
  id: "payment_i2NJhL",
};

describe("Payment", () => {
  beforeAll(() => {
    mockEndpoint({
      url: "/payments/*",
      response: mockPaymentJSON,
      status: 200,
      method: MockableHTTPMethods.GET,
    });
  });
  it("should render without crashing", () => {
    const { container } = render(
      withProviders(<Payment id={mockPayment.id} />)
    );
    expect(container).toMatchSnapshot();
  });
});
