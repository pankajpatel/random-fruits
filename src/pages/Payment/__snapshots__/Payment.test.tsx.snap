// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should render without crashing 1`] = `
.c4 {
  border: 2px solid transparent;
  border-left-color: #333;
  border-radius: 50%;
  width: 1.2rem;
  height: 1.2rem;
  display: block;
  margin: 0.25rem;
  -webkit-animation: keep-spinning 500ms infinite;
  animation: keep-spinning 500ms infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 0 0.5rem;
  margin-bottom: 1.5rem;
}

.c3 {
  font-size: 1.2rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.25rem;
  width: 100%;
}

.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > * {
  margin-right: 0.5rem;
}

.c2 {
  border: 1px solid rgba(0,0,0,0.2);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding: 0.5rem;
  display: inline-block;
  border-radius: 0.25rem;
}

<div>
  <section>
    <header
      class="c0"
    >
      <h2
        class="c1"
      >
        <a
          class="c2"
          href="/payments"
        >
          <svg
            height="1rem"
            version="1.1"
            viewBox="0 0 50.999 85.999"
            width="1rem"
          >
            <path
              d="M44.998,80.094c1.338,1.352,1.338,3.541,0,4.893c-1.336,1.35-3.506,1.352-4.844,0L1.003,45.447  c-1.338-1.352-1.338-3.543,0-4.895l39.15-39.539c1.338-1.352,3.506-1.352,4.844,0S46.335,4.555,45,5.906L9.294,43L44.998,80.094z"
            />
          </svg>
        </a>
        <span>
          page.payment.title
        </span>
      </h2>
      <div
        class=""
      >
        <select
          class="c3"
        >
          <option
            value="en"
          >
            EN ðŸ‡¬ðŸ‡§
          </option>
          <option
            value="fr"
          >
            FR ðŸ‡«ðŸ‡·
          </option>
        </select>
      </div>
    </header>
    <div
      class="c4"
      data-testid="spinner"
      title="Loading..."
    />
  </section>
</div>
`;

exports[`Payment should render without crashing 2`] = `
.c6 {
  display: block;
  font-size: 1.2rem;
  color: #aaa;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  border-bottom: 1px solid rgba(0,0,0,0.2);
  padding: 0 0.5rem;
  margin-bottom: 1.5rem;
}

.c3 {
  font-size: 1.2rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 0.25rem;
  width: 100%;
}

.c1 {
  margin-top: 1rem;
  margin-bottom: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > * {
  margin-right: 0.5rem;
}

.c8 {
  padding: 0.35em 0.55em;
  border-radius: 0.4em;
  background-color: #e3f2fd;
  border: 1px solid rgba(0,0,0,0.05);
  line-height: 1;
}

.c18 {
  padding: 0.35em 0.55em;
  border-radius: 0.4em;
  background-color: #fafafa;
  border: 1px solid rgba(0,0,0,0.05);
  line-height: 1;
}

.c4 {
  padding: 0.5rem;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.c4 .c5 + span {
  font-size: 2rem;
  display: block;
  margin-bottom: 1.5rem;
}

.c4 .c7 {
  font-size: 1rem;
}

.c14 {
  cursor: pointer;
  font-size: 1.1rem;
}

.c14:nth-child(even) {
  background-color: #fafafa;
}

.c14:hover {
  background-color: #eee;
}

.c16 {
  padding: 0.75rem 0.5rem;
  text-align: left;
}

.c16 .c5 {
  display: none;
}

.c17 {
  padding: 0.75rem 0.5rem;
  text-align: right;
}

.c17 .c5 {
  display: none;
}

.c10 {
  border-collapse: collapse;
  width: 100%;
}

.c11 {
  padding: 0.75rem 0.5rem;
  text-align: left;
  background-color: #eee;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-style: dotted;
  text-decoration-style: dotted;
}

.c11:hover {
  background-color: #ddd;
}

.c12 {
  padding: 0.75rem 0.5rem;
  text-align: right;
  background-color: #eee;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-style: dotted;
  text-decoration-style: dotted;
}

.c12:hover {
  background-color: #ddd;
}

.c9 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  border: 1px solid rgba(0,0,0,0.2);
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  cursor: pointer;
  padding: 0.5rem;
  display: inline-block;
  border-radius: 0.25rem;
}

@media (max-width:768px) {
  .c10 {
    display: grid;
  }

  .c10 thead {
    display: none;
  }

  .c10 tbody {
    display: grid;
  }

  .c10 .c13 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 0.5rem;
    padding: 0.5rem;
  }

  .c10 .c15 {
    text-align: left;
    padding: 0;
  }

  .c10 .c15 .c5 {
    display: block;
    font-size: 0.75em;
    line-height: 1.2;
  }
}

<div>
  <section>
    <header
      class="c0"
    >
      <h2
        class="c1"
      >
        <a
          class="c2"
          href="/payments"
        >
          <svg
            height="1rem"
            version="1.1"
            viewBox="0 0 50.999 85.999"
            width="1rem"
          >
            <path
              d="M44.998,80.094c1.338,1.352,1.338,3.541,0,4.893c-1.336,1.35-3.506,1.352-4.844,0L1.003,45.447  c-1.338-1.352-1.338-3.543,0-4.895l39.15-39.539c1.338-1.352,3.506-1.352,4.844,0S46.335,4.555,45,5.906L9.294,43L44.998,80.094z"
            />
          </svg>
        </a>
        <span>
          page.payment.title
        </span>
      </h2>
      <div
        class=""
      >
        <select
          class="c3"
        >
          <option
            value="en"
          >
            EN ðŸ‡¬ðŸ‡§
          </option>
          <option
            value="fr"
          >
            FR ðŸ‡«ðŸ‡·
          </option>
        </select>
      </div>
    </header>
    <div
      class="c4"
    >
      <div>
        <label
          class="c5 c6"
        >
          label.customer
        </label>
        <span>
          Ã‰mile Zola
        </span>
      </div>
      <div>
        <label
          class="c5 c6"
        >
          label.merchant
        </label>
        <span>
          Super Merchant
        </span>
      </div>
      <div>
        <label
          class="c5 c6"
        >
          label.amount
        </label>
        <span>
          356050
           â‚¬
        </span>
      </div>
      <div>
        <label
          class="c5 c6"
        >
          label.created
        </label>
        <span>
          Mon Apr 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
        </span>
      </div>
      <div>
        <label
          class="c5 c6"
        >
          label.status
        </label>
        <span>
          <span
            class="c7 c8"
            type="info"
          >
            status.ready
          </span>
        </span>
      </div>
    </div>
    <div
      class="c9"
    >
      <label
        class="c5 c6"
      >
        page.payment.installments.label
      </label>
      <table
        class="c10"
      >
        <thead>
          <tr>
            <th
              class="c11"
            >
              table.header.dueDate
              <small>
                â‡µ
              </small>
            </th>
            <th
              class="c12"
            >
              table.header.amount
              <small>
                â‡µ
              </small>
            </th>
            <th
              class="c12"
            >
              table.header.fee
              <small>
                â‡µ
              </small>
            </th>
            <th
              class="c12"
            >
              table.header.status
              <small>
                â‡µ
              </small>
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Mon Apr 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              500
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c8"
                type="info"
              >
                status.ready
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Wed May 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Sat Jun 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Mon Jul 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Thu Aug 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Sun Sep 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Tue Oct 04 2022 00:00:00 GMT+0200 (Central European Summer Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Fri Nov 04 2022 00:00:00 GMT+0100 (Central European Standard Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Sun Dec 04 2022 00:00:00 GMT+0100 (Central European Standard Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
          <tr
            class="c13 c14"
            title="Click to open payment details"
          >
            <td
              class="c15 c16"
            >
              <label
                class="c5 c6"
              >
                table.header.dueDate
              </label>
              Wed Jan 04 2023 00:00:00 GMT+0100 (Central European Standard Time)
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.amount
              </label>
              35555
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.fee
              </label>
              0
               â‚¬
            </td>
            <td
              class="c15 c17"
            >
              <label
                class="c5 c6"
              >
                table.header.status
              </label>
              <span
                class="c7 c18"
                type="default"
              >
                status.not_started
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;
